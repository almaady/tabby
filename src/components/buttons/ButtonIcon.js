import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import Icon from "../icons/Icon";

/**
 * @class ButtonIcon - It is a button that works as an icon, without text
 */
class ButtonIcon extends React.PureComponent {
  /**
   * @property {string} icon -The name of the icon
   * @property {string} color - The color of the icon
   * @property {string} size- The size of the icon
   * @property {func} onClick - The function triggered by the button
   * @property {string} className - Just in case you need another class
   * @property {string} type - Button type
   * @property {bool} disabled - Button could be disabled
   * @property {string} borderColor - Button could have a border color
   */
  static propTypes = {
    icon: PropTypes.string.isRequired,
    color: PropTypes.string,
    size: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string,
    type: PropTypes.string,
    disabled: PropTypes.bool,
    borderColor: PropTypes.string
  };

  render() {
    const { icon, color, size, onClick, className, type, disabled, borderColor } = this.props;
    return (
      <button
        type={type}
        className={classNames("btn", "btn--icon",`border--${borderColor}`, className)}
        onClick={onClick}
        disabled={disabled}
      >
        <Icon icon={icon} color={color} size={size} />
      </button>
    );
  }
}
ButtonIcon.defaultProps = {
  className: "",
  size: "XXL",
  color: "gray-l1",
  type: "button",
  disabled: false,
  borderColor: ""
};

export default ButtonIcon;
