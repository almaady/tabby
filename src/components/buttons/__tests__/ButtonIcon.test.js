import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";

import ButtonIcon from "../ButtonIcon";

describe("ButtonIcon", () => {
  const action = sinon.stub();
  let props;
  beforeEach(() => {
    props = {
      icon: "cross",
      onClick: action
    };
  });
  it("should render correctly with default state", () => {
    const component = shallow(<ButtonIcon {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should call action when you click ", () => {
    const component = shallow(<ButtonIcon {...props} />);
    component.simulate("click");
    expect(action.called).toBeTruthy();
  });
  it("should render new className", () => {
    const component = shallow(<ButtonIcon {...props} className="btn-2" />);
    expect(component.hasClass("btn-2"));
  });
});
