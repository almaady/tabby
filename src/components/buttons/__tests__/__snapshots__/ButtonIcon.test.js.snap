// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonIcon
    borderColor=""
    className=""
    color="gray-l1"
    disabled={false}
    icon="cross"
    onClick={[Function]}
    size="XXL"
    type="button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Icon
        color="gray-l1"
        icon="cross"
        size="XXL"
      />,
      "className": "btn btn--icon border--",
      "disabled": false,
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "gray-l1",
        "icon": "cross",
        "size": "XXL",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Icon
          color="gray-l1"
          icon="cross"
          size="XXL"
        />,
        "className": "btn btn--icon border--",
        "disabled": false,
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "gray-l1",
          "icon": "cross",
          "size": "XXL",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
